src/sinks/util/encoding.rs

```diff
@@ -156,7 +159,23 @@ pub trait EncodingConfiguration<E> {
                 Event::Log(log_event) => {
                     let to_remove = log_event
                         .keys()
-                        .filter(|f| !only_fields.contains(f))
+                        .filter(|field| {
+                            let field_path = PathIter::new(field).collect::<Vec<_>>();
+                            !only_fields.iter().any(|only| {
+                                if only == field {
+                                    true // fastpath
+                                } else {
+                                    // TODO: This is a hack for #2407, #2410 should fix this fully.
+                                    // This is *slow*.
+                                    let only_path = PathIter::new(only);
```

Comment:
If we're worried about perf here, could we do the iter and collect once at startup?

Reply:
This seems worth it if true.

Reply:
Yeah!!!! I was thinking about doing it right at startup time of the `EncodingConfiguration` itself, but thought that would be a bit invasive. Let me go ahead and just do it then.